packages/service/tests/unit/test_discovery_client.py

```diff
@@ -393,3 +394,7 @@ def _assert_service_info_equal(
     assert set(expected.provided_interfaces) == set(actual.provided_interfaces)
     assert expected.service_class == actual.service_class
     assert expected.annotations == actual.annotations
+    if isinstance(actual, GrpcServiceDescriptor):
+        assert expected.version == actual.versions[0]
+    else:
+        assert expected.version == actual.version
```

Comment:
This highlights an impending conflict with #826. You are adding a singular `version` field to `ServiceInfo`, but @Jotheeswaran-Nandagopal is adding support for `enumerate_services`. According to the API docs, `EnumerateServicesResponse` contains a list of versions.

```
  // This field is used differently based on the message in which it is used.
  //
  // When used in a RegisterServiceRequest, this field must contain a single
  // version, and that version must follow the major.minor.patch format.
  // The version 0.0.1 is reserved for internal use, and cannot be specified.
  // If no version is specified, the version 0.0.1 will be used internally
  // for the registered service.
  //
  // When returned from EnumerateServicesResponse, this field will contain the
  // list of versions that are available for the associated service.
  //
  // When returned from ResolveServiceWithInformationResponse, this field will
  // contain the version of the resolved service.
  repeated string versions = 6;
```

Reply:
Sorry - I reset you without seeing this message.  Thanks for pointing this out - I think it makes sense for ServiceInfo to have `versions` rather than `version`.  I'll make the update.